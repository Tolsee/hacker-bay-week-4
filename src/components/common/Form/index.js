// @flow
import * as React from 'react';
import styled from 'styled-components';
import {
  Input,
  Alert
} from 'antd';

export const ErrorMessage = styled.span`
  color: ${({ theme }) => theme.danger };
  line-height: 1.5;
  -webkit-transition: color .3s cubic-bezier(.215,.61,.355,1);
  transition: color .3s cubic-bezier(.215,.61,.355,1);
  margin-top: -2px;
  clear: both;
`;

// Field with message
type fieldProps = {
  input: Object ,
  meta: { error: any },
  prefix: React.Node,
  suffix: React.Node,
  type: string,
  label: string,
  fieldTouched: boolean
};

export const renderField =  ({
                               input ,
                               meta: { error },
                               prefix,
                               suffix,
                               type,
                               label,
                               fieldTouched
                             }: fieldProps)  =>  (
  <React.Fragment>
    <Input
      {...input}
      placeholder={label}
      prefix={prefix}
      Suffix={suffix}
      type={type}
    />
    { fieldTouched && error && <ErrorMessage>{error}</ErrorMessage>}
  </React.Fragment>
);

export const Info = styled(Alert)`
  && {
    margin: 10px 0;
  }
`;

