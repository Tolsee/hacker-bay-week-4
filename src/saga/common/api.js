import { message } from 'antd';

const a = 1;
const b = 2;

export const apiOnSuccess = msg => function* onSuccess() {
  message.success(msg);
};

export const apiOnError = function* onError(action) {
  message.error(action.error);
};