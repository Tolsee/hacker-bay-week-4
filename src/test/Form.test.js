import React from 'react';
import { shallow } from 'enzyme';
import { Info } from 'components/common/Form';

it('Info renders correctly', () => {
  const component = shallow(<Info message="Error message" type="error" />);

  expect(component).toMatchSnapshot();
  expect(component.prop('message')).toEqual('Error message');
});