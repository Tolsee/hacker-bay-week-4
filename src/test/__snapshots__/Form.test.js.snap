// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info renders correctly 1`] = `
<Alert
  className="sc-bwzfXH hQDmqP"
  message="Facebook"
  type="error"
/>
`;
